apps/hannk/interpreter/tensor.h

```diff
@@ -78,12 +86,13 @@ class Tensor {
     // for a Tensor to be dynamic if it is also constant or external.
     // Currently only used in conjunction with the TFLite delegate.
     bool is_dynamic_ = false;
-    // If true, this Tensor shares its TensorStorage with at least one other Tensor.
-    bool is_alias_ = false;
-    // If true, this Tensor is aliased to another via reshape (rather than cropping),
-    // and may have a different rank from the underlying storage.
-    // Only used if is_alias_ = true.
-    bool is_reshape_alias_ = false;
+
+    // All the Tensors in this list share their TensorStorage with each other.
+    // An arbitrary number of Tensors can alias each other, but at most *one* may be external.
+    //
+    // Note that this list should never have size() of 0 or 1 (the pointer should simply be null).
+    std::shared_ptr<std::vector<std::weak_ptr<Tensor>>> aliases_;
+    AliasType alias_type_ = AliasType::None;
```

Comment:
Shouldn't this be inside the shared_ptr above? A tuple/struct with the vector?

Reply:
Hm, yes, that is probably a better design, will do.